test/dlc_tests/sanity/test_ecr_scan.py

```diff
@@ -272,6 +275,29 @@ def helper_function_for_leftover_vulnerabilities_from_enhanced_scanning(
         LOGGER.info(
             f"[NonPatchableVulns] [image_uri:{ecr_enhanced_repo_uri}] {json.dumps(non_patchable_vulnerabilities.vulnerability_list, cls= test_utils.EnhancedJSONEncoder)}"
         )
+
+    if not is_generic_image():
```

Comment:
Can we add a logic to make sure that this push only happens in the build pipelines. You can do that by calling this function: https://github.com/aws/deep-learning-containers/blob/master/test/test_utils/__init__.py#L679.

We can restrict the uploads to only when this function is true.